dev/TEAL_opcodes.md

```diff
@@ -1128,3 +1229,74 @@ bitlen interprets arrays as big-endian integers, unlike setbit/getbit
 - Pushes: []byte
 - push a byte-array of length X, containing all zero bytes
 - LogicSigVersion >= 4
+
+## log
+
+- Opcode: 0xb0
+- Pops: *... stack*, []byte
+- Pushes: _None_
+- write bytes to log state of the current application
+- LogicSigVersion >= 5
+- Mode: Application
+
+`log` can be called up to MaxLogCalls times in a program, and log up to a total of 1k bytes.
+
+## tx_begin
+
+- Opcode: 0xb1
+- Pops: _None_
+- Pushes: _None_
+- Begin preparation of a new inner transaction
+- LogicSigVersion >= 5
+- Mode: Application
+
+## tx_field f
+
+- Opcode: 0xb2 {uint8 transaction field index}
+- Pops: *... stack*, any
+- Pushes: _None_
+- Set field F of the current inner transaction to X
+- LogicSigVersion >= 5
+- Mode: Application
+
+## tx_submit
```

Comment:
Do we want a `tx_try`?
If yes, should this keep the current transaction available for `tx_submit` or does the program needs to reconstruct it?

Reply:
I don't intend to put it in the one.  I guess I would say tx_try will, someday, try to perform the txn, and return success/fail on the stack.  So if it can run, it does run.  So it won't be needed to stay around.  But I don't know for sure.